src/Symfony/Component/Security/Core/Encoder/EncoderFactory.php

```diff
@@ -175,4 +172,22 @@ private function getEncoderConfigFromAlgorithm(array $config): array
             ],
         ];
     }
+
+    private function createMigratingEncoder(array $config, array $algos)
+    {
+        foreach ($algos as $algo) {
+            // "plaintext" is not allowed as any leaked hashes could then be used to authenticate directly
+            if ('plaintext' === $algo) {
+                throw new LogicException('Migrating plaintext passwords is not supported.');
+            }
+
+            $config['algorithm'] = $algo;
```

Comment:
this assumes that all other settings are the same for all algos, which look wrong to me

Reply:
fixed